xmppserver/src/main/java/org/jivesoftware/openfire/ldap/LdapGroupProvider.java

```diff
@@ -124,7 +133,67 @@ private Group getGroupByDN(LdapName groupDN, Set<String> membersToIgnore) throws
             ctx = manager.getContext(baseDN);
             Attributes attrs = ctx.getAttributes(relativeDN, standardAttributes);
 
-            return processGroup(ctx, attrs, membersToIgnore);
+            boolean paging = false;
+            int countPage=0;
+
+            NamingEnumeration<? extends Attribute> i = attrs.getAll();
+            String groupMemberField = manager.getGroupMemberField();
+            String searchString = groupMemberField.contains(";range=")?groupMemberField.substring(0,groupMemberField.indexOf(";range=")):groupMemberField;
+            while (i.hasMore())
+            {
+                Attribute attribute = i.next();
+                String key = attribute.getID().toLowerCase();
+                if (key.startsWith(searchString))
+                {
+                    paging=true;
+                    break;
+                }
+            }
+
+            if (!paging)
+            {
+                return processGroup(ctx, attrs, membersToIgnore);
+            }
+            else
+            {
+                //retrieve first 0-1499
+                Group tmpGroup = processGroup(ctx, attrs, membersToIgnore);
+                Collection<JID> admins = tmpGroup.getAdmins();
+                String description = tmpGroup.getDescription();
+                String name = tmpGroup.getName();
+                ArrayList<JID> members = new ArrayList<JID>();
+                members.addAll(tmpGroup.getMembers());
+                do
+                {
+                    try
+                    {
+                        ArrayList<String> stdAttr = new ArrayList<String>();
+                        for (int n=0;n<standardAttributes.length;n++)
+                        {
+                            if (!standardAttributes[n].contains(manager.getGroupMemberField()))
```

Comment:
Style: please always wrap if-blocks in `{` and `}` even if they're just one line. Mistakes can be made easily without them.